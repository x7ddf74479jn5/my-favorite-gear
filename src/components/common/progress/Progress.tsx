import CircularProgress from "@mui/material/CircularProgress";
import Container from "@mui/material/Container";
import Typography from "@mui/material/Typography";
import type { VFC } from "react";
import React from "react";

const Progress: VFC = () => {
  return (
    <Container
      sx={{ textAlign: "center", margin: (theme) => theme.spacing(4, 0) }}
    >
      <Typography>読み込み中</Typography>
      <CircularProgress size={50} />
    </Container>
  );
};

export default Progress;
