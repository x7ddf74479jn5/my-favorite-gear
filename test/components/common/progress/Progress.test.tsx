import React from "react";
import { describe, expect, it } from "vitest";

import Progress from "@/components/common/progress/Progress";

import { render } from "../../../test-utils";

describe("Progress", () => {
  it("matches snapshot", () => {
    const renderResult = render(<Progress />);

    expect(renderResult.asFragment()).toMatchSnapshot();
  });

  it("renders correctly", () => {
    const renderResult = render(<Progress />);
    expect(renderResult.container.firstChild).toHaveAttribute(
      "class",
      expect.stringContaining("styledContainer")
    );

    expect(renderResult.getByText("読み込み中")).toBeInTheDocument();
    expect(renderResult.container.getElementsByTagName("svg")).toBeTruthy();
    expect(renderResult.container.getElementsByTagName("circle")).toBeTruthy();
  });
});
