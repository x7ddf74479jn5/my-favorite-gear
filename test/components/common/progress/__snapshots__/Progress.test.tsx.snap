// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root makeStyles-styledContainer-1 MuiContainer-maxWidthLg"
  >
    <p
      class="MuiTypography-root MuiTypography-body1"
    >
      読み込み中
    </p>
    <div
      class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 50px; height: 50px;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
